packages/react-router-dom/dom.ts

```diff
@@ -243,27 +253,30 @@ export function getFormSubmissionInfo(
       `Cannot submit element that is not <form>, <button>, or ` +
         `<input type="submit|image">`
     );
-  } else {
+  } else if (target instanceof FormData) {
     method = options.method || defaultMethod;
     action = options.action || null;
     encType = options.encType || defaultEncType;
-
-    if (target instanceof FormData) {
-      formData = target;
-    } else {
-      formData = new FormData();
-
-      if (target instanceof URLSearchParams) {
-        for (let [name, value] of target) {
-          formData.append(name, value);
-        }
-      } else if (target != null) {
-        for (let name of Object.keys(target)) {
-          formData.append(name, target[name]);
-        }
-      }
+    formData = target;
+  } else if (target instanceof URLSearchParams) {
+    method = options.method || defaultMethod;
+    action = options.action || null;
+    encType = options.encType || defaultEncType;
+    formData = new FormData();
+    for (let [name, value] of target) {
+      formData.append(name, value);
     }
+  } else {
+    // Anything left at this point is a raw payload and router will serialize
+    // it into the request based on the encType
+    method = options.method || defaultMethod;
+    action = options.action || null;
+    // In v7 we want to remove the defaultEncType from here so it defaults to
+    // no serialization
+    encType =
+      options.encType === null ? null : options.encType || defaultEncType;
+    payload = target;
```

Comment:
@ryanflorence Just a call out that this felt cleaner but breaks mutual exclusivity of `payload`/`request.formData`.  IMO this aligned better with the handling of `application/json` and `text/plain`.  

Basically, _any_ raw `payload` (non-`HTMLElement`, non-`FormData`, non-`URLSearchParams`) will always show up to the `action` as `payload`.  Then, based on the `encType` it might _also_ be in `request.formData|json|text`

```js
submit(obj, { encType: "application/x-www-form-urlencoded" })
// action payload === obj, serialized into request.formData

// Default behavior is the same as above
submit(obj)
// action payload === obj, serialized into request.formData

submit(obj, { encType: "application/json" })
// action payload === obj, serialized into request.json

submit("some text", { encType: "text/plain" })
// action payload === "some text", serialized into request.text

submit(obj, { encType: null })
// action payload === obj, not serialized
```